app/views/pages/use-cases/using-websockets.liquid

```diff
@@ -0,0 +1,195 @@
+
+
+### Step 2: Creating a connection
+
+Consumers require an instance of the connection on their side. This can be established using the code below.
+
+```javascript
+// modules/chat/src/js/consumer.js
+
+import { createConsumer } from "@rails/actioncable"
+
+export default createConsumer('/websocket')
+```
+
+### Step 3: Subscribing a channel
+
+A consumer becomes a subscriber by creating a subscription to a given channel:
+
+```javascript
+// modules/chat/src/js/chat.js
+import consumer from "./consumer";
+
+const chat = function(){
+
+  // cache 'this' value not to be overwritten later
+  const module = this;
+
+  module.createSubscription = () => {
+      module.channel = consumer.subscriptions.create(
+      {
+        channel: 'conversate', //channel name
+        room_id: module.conversationId, //room ID
+      }
+    )
+  }
+}
+```
+
+### Step 4: Handling callbacks
+
+There are 4 callbacks handled by the subscription that allows us to react on certain events, we can add handlers for them when we create subscription object:
+
+```javascript
+// modules/chat/src/js/chat.js
+import consumer from "./consumer";
+
+const chat = function(){
+
+  // cache 'this' value not to be overwritten later
+  const module = this;
+  module.createSubscription = () => {
+    module.channel = consumer.subscriptions.create(
+      {
+        channel: 'conversate',
+        room_id: module.conversationId
+      },
+      {
+        received: function(data){
+          // function responsible for handling new message
+          module.showMessage(data);
+        },
+
+        connected: function(){
+          // on connect we want to enable the message input
+          module.settings.messageInput.disabled = false;
+          module.settings.messageInput.focus();
+        },
+
+        rejected: function(){
+          // when connection has been rejected we want to notify the user about the error
+          module.blocked();
+        },
+
+        disconnected: function(){
+          // when consumer has been disconnected we want to notify the user about the error
+          module.blocked();
+        }
+      }
+    );
+  };
+}
+```
+
+### Step 5: Securing room subscription
+
+To make sure that only authorized users can access certain rooms (for example rooms for conversations where they are one of the pariticpants) we can use a partial that will be called when user tries to subscribe to a certain room.
+This file should be stored in `views/partials/channels/:channel_name/subscribed.liquid`. In this example we use code from the [pOS Marketplace Template](/developer-guide/pos-marketplace-template/core-concepts) and there channel is called `conversate`. This partial needs to return `true` string if current user is authorized to subscribe the room.
```

Comment:
In `views/partials/channels/:channel_name/subscribed.liquid`, is the `:` needed?

Reply:
its variable, that's why the `:`